Project/DAQ_System/Program/Src/Application/data_payload.hpp

```diff
@@ -37,32 +42,122 @@ struct DataPayload {
     float tps_ = 0.0f;
     float fuel_open_time_ = 0.0f;
     float ignition_angle_ = 0.0f;
+
+    float barometer_ = 0.0f;
     float map_ = 0.0f;
     float lambda_ = 0.0f;
+
+    static constexpr uint8_t kNumAnalogFields = 8;
+	std::array<float, kNumAnalogFields> analog_inputs_{};
+
     float battery_voltage_ = 0.0f;
     float air_temp_ = 0.0f;
     float coolant_temp_ = 0.0f;
 
     int GetCsvFormatLength() {
-		return snprintf(NULL, 0,
-				"%f,%f,%f,%f,%f\n",
-				timestamp_,
-				linpot_displacement_inches_[0],
-				linpot_displacement_inches_[1],
-				linpot_displacement_inches_[2],
-				linpot_displacement_inches_[3]);
+    	return CsvString(NULL, 0);
 	}
 
     void CsvFormat(char* buffer, int length) {
-		snprintf(buffer, length,
-				"%f,%f,%f,%f,%f\n",
-				timestamp_,
-				linpot_displacement_inches_[0],
-				linpot_displacement_inches_[1],
-				linpot_displacement_inches_[2],
-				linpot_displacement_inches_[3]);
+    	CsvString(buffer, length);
 	}
 
+    void Lock() { mutex_->Lock(); }
+
+    void Unlock() { mutex_->Unlock(); }
+
+    static constexpr char* kCsvHeaderRow = (char*)"Timestamp,"
+
+    								   			  "Linpot1,"
+    								   			  "Linpot2,"
+    								   			  "Linpot3,"
+    								   			  "Linpot4,"
+
+    								   			  "AccX,"
+    								   			  "AccY,"
+    								   			  "AccZ,"
+
+    								   			  "GyroX,"
+    								   			  "GyroY,"
+    								   			  "GyroZ,"
+
+    								   			  "RPM,"
+    								   			  "TPS,"
+    								   			  "FuelOpenTime,"
+    								   			  "IgnitionAngle,"
+
+    								   			  "Barometer,"
+    								   			  "MAP,"
+    								   			  "Lambda,"
+
+    								   			  "AnalogInput1,"
+    								   			  "AnalogInput2,"
+    								   			  "AnalogInput3,"
+    								   			  "AnalogInput4,"
+    								   			  "AnalogInput5,"
+    								   			  "AnalogInput6,"
+    								   			  "AnalogInput7,"
+    								   			  "AnalogInput8,"
+
+    								   			  "BatteryVoltage,"
+    								   			  "AirTemp,"
+    								   			  "CoolantTemp"
+    								   			  "\n\0";
```

Comment:
Would it maybe be useful to give the user a way to rename these?

Reply:
The concept of user settings warrants saving data that will persist even after the device is shutdown — we need non-volatile memory, such as EEPROM to store the settings.

Also, Software is having to sync up with the format for parsing the CSV file.

So I don't think that configurability is desirable as of now.